tests/test_apply.py

```diff
@@ -36,3 +39,120 @@ def test_apply(temporal_interval, bounding_box, random_raster_data, process_regi
     )
 
     xr.testing.assert_equal(output_cube, input_cube + 1)
+
+
+@pytest.mark.parametrize("size", [(6, 5, 4, 4)])
+@pytest.mark.parametrize("dtype", [np.float32])
+def test_apply_dimension_case_1(
+    temporal_interval, bounding_box, random_raster_data, process_registry
+):
+    input_cube = create_fake_rastercube(
+        data=random_raster_data,
+        spatial_extent=bounding_box,
+        temporal_extent=temporal_interval,
+        bands=["B02", "B03", "B04", "B08"],
+        backend="dask",
+    )
+
+    _process = partial(
+        process_registry["add"], y=1, x=ParameterReference(from_parameter="data")
```

Comment:
The new `apply_dimension` looks really good and I think these tests are fine, but we should make sure that this also works with the `order` , `rearrange` and `sort` process as these cannot be used with `apply` or `reduce_dimension`.

Reply:
Will look into that in the other PR!